lib/filterx/object-datetime.c

```diff
@@ -200,6 +200,64 @@ _repr(FilterXObject *s, GString *repr)
   return TRUE;
 }
 
+FilterXObject *
+filterx_datetime_strptime(GPtrArray *args)
+{
+  if (args == NULL || args->len < 2)
+    {
+      msg_error("FilterX: Failed to create datetime object: invalid number of arguments. "
+                "Usage: strptime(time_str, format_str0, ..., format_strN)");
+      return NULL;
+    }
+
+  FilterXObject *object = g_ptr_array_index(args, 0);
+  if (!object || !filterx_object_is_type(object, &FILTERX_TYPE_NAME(string)))
```

Comment:
Please also handle and test `message_ref`.

This filterx block fails:
```
  filterx {
    $date = strptime($int_from_msg, "%s");
  };
```

Reply:
yeah, either that or we would need to call unmarshall (or eval_typed). we might need to have a function where all arguments are automatically unmarshalled?

Reply:
We can save the unmarshal (alloc and copy), if we check that the message value is string or not. This is what Shifi did.